src/main/java/com/mongodb/spark/sql/connector/config/WriteConfig.java

```diff
@@ -0,0 +1,238 @@
+   * provide the number of MongoDB's required to acknowledge the write before continuing.
+   *
+   * <p>Note: The default write concern is {@code WriteConcern.ACKNOWLEDGED}.
+   */
+  public static final String WRITE_CONCERN_W_CONFIG = "writeConcern.w";
+
+  /**
+   * The optional {@link WriteConcern} journal property.
+   *
+   * <p>Note: Must be a boolean string: {@code true} or {@code false}.
+   */
+  public static final String WRITE_CONCERN_JOURNAL_CONFIG = "writeConcern.journal";
+
+  /**
+   * The optional {@link WriteConcern} wTimeout property in milliseconds.
+   *
+   * <p>Note: Must be a valid integer
+   */
+  public static final String WRITE_CONCERN_W_TIMEOUT_MS_CONFIG = "writeConcern.wTimeoutMS";
+
+  private final WriteConcern writeConcern;
+  private final OperationType operationType;
+
+  /**
+   * Construct a new instance
+   *
+   * @param options the options for configuration
+   */
+  WriteConfig(final Map<String, String> options) {
+    super(options, UsageMode.WRITE);
+    writeConcern = createWriteConcern();
+    operationType =
+        OperationType.fromString(getOrDefault(OPERATION_TYPE_CONFIG, OPERATION_TYPE_DEFAULT));
+  }
+
+  /**
+   * Return a {@link WriteConfig} instance with the extra options.
+   *
+   * <p>Existing configurations may be overwritten by the new options.
+   *
+   * @param overrides the context specific options.
+   * @return the WriteConfig with the overrides
+   */
+  @Override
+  public WriteConfig withOptions(final Map<String, String> overrides) {
+    if (overrides.isEmpty()) {
+      return this;
+    }
+    return new WriteConfig(withOverrides(overrides));
+  }
+
+  /** @return the max size of bulk operation batches */
+  public int getMaxBatchSize() {
+    return getInt(MAX_BATCH_SIZE_CONFIG, MAX_BATCH_SIZE_DEFAULT);
+  }
+
+  /** @return the operation type */
+  public OperationType getOperationType() {
+    return operationType;
+  }
+
+  /** @return the write concern to sue */
+  public WriteConcern getWriteConcern() {
+    return writeConcern;
+  }
+
+  /** @return the field list used to identify the document */
+  public List<String> getIdFields() {
+    return getList(ID_FIELD_CONFIG, ID_FIELD_DEFAULT);
+  }
+
+  /** @return true if the bulk operation is ordered */
+  public boolean isOrdered() {
+    return getBoolean(ORDERED_BULK_OPERATION_CONFIG, ORDERED_BULK_OPERATION_DEFAULT);
+  }
+
+  private WriteConcern createWriteConcern() {
+    WriteConcern writeConcern = WriteConcern.ACKNOWLEDGED;
+    try {
+      if (containsKey(WRITE_CONCERN_W_CONFIG)) {
+        try {
+          writeConcern = writeConcern.withW(getInt(WRITE_CONCERN_W_CONFIG, -1));
+        } catch (NumberFormatException e) {
+          // ignore
+          writeConcern = writeConcern.withW(get(WRITE_CONCERN_W_CONFIG));
+        }
+      }
+
+      if (containsKey(WRITE_CONCERN_JOURNAL_CONFIG)) {
+        writeConcern = writeConcern.withJournal(getBoolean(WRITE_CONCERN_W_CONFIG, false));
```

Comment:
Thanks good catch - I will add unit tests for the configs in SPARK-313 I will add a todo in the code

Reply:
[optional]
`TODO` comments left in code have a tendency to stay there forever (the Java driver repo has TODO's since at least 2012). I think `TODO` are a legit way to leave bookmarks only within the work-in-progress branches, and should not ever be put into release branches. Please consider leaving the notes in the appropriate Jira ticket, rather than committing `TODO`'s into a release branch. The same is applicable to the `TODO` comment in `AbstractMongoConfig.java`. Ross, if you disagree with this proposal, don't hesitate to resolve the discussion.